lib/stdlib/test/proplists_SUITE.erl

```diff
@@ -77,4 +77,83 @@ examples(_Config) ->
     NegListRes = [{foo, false}, {foo, false}, foo, foo, foo],
     NegListRes = proplists:substitute_negations([{no_foo, foo}], NegList),
 
+    true = #{a => true, b => 1, c => 2} =:= proplists:to_map([a, {b, 1}, {c, 2}, {c, 3}]),
+
+    ok.
+
+map_conversion(_Config) ->
+    %% Simple tests.
+    true = #{} =:= proplists:to_map([]),
+    true = #{a => true, b => true} =:= proplists:to_map([a, b]),
+    true = #{a => true, b => true} =:= proplists:to_map([b, a]),
+    true = #{a => 1, b => true} =:= proplists:to_map([{a, 1}, b]),
+    true = #{a => 1, b => true} =:= proplists:to_map([b, {a, 1}]),
+    true = #{a => 1, b => 2} =:= proplists:to_map([{a, 1}, {b, 2}]),
+    true = #{a => 1, b => 2} =:= proplists:to_map([{b, 2}, {a, 1}]),
+    true = #{b => true} =:= proplists:to_map(["a", b]),
+    true = #{b => true} =:= proplists:to_map([b, "a"]),
+    true = #{b => true} =:= proplists:to_map([{a}, b]),
+    true = #{b => true} =:= proplists:to_map([b, {a}]),
+    true = #{b => true} =:= proplists:to_map([{a, 1, 2}, b]),
+    true = #{b => true} =:= proplists:to_map([b, {a, 1, 2}]),
+
+    %% Ensure that maps:get/3 using the created map yields the same
+    %% results as proplists:get_value/3 on the original proplist does,
+    %% and that proplists:get_value/3 on a proplist created from the
+    %% map yields the same results as proplists:get_value/3 on the
+    %% original proplist, ie they either all return the same `Value',
+    %% or they all return the `Default' given as respective third argument.
+    Default1 = make_ref(),
+    Default2 = make_ref(),
+    Default3 = make_ref(),
+    InList=[a, b, {a, 1}, {}, {a}, {a, 1, 2}, "foo"],
```

Comment:
I think a `{c}` or `{c,1,2}` term in the list would be nice, i.e a tuple with a key that has no 2-tuple that can shadow it.

Reply:
Ok, will do :) I had more in there, but the number of variations is `n!` (with `n` being the number of items), so I had to cut back a little ;) But one more is ok I guess.

Reply:
Ouch!  Ok.  It could maybe replace the `{}` term if the size is a problem.

Reply:
Ok, I just added `{c, 1, 2}`, the addition raises then number of combinations from 5040 to 40320, but that's still ok I think.